Formula/gomigrate.rb

```diff
@@ -0,0 +1,26 @@
+class Gomigrate < Formula
+  desc "Database migrations CLI tool"
+  homepage "https://github.com/golang-migrate/migrate"
+  url "https://github.com/golang-migrate/migrate/archive/v3.3.0.tar.gz"
+  sha256 "59ade052631d6a064d3ac71df5f03c120c3029210d8d13972b6c5db7558c490f"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/golang-migrate/migrate").install buildpath.children
+
+    # Build and install cli as "migrate"
+    cd "src/github.com/golang-migrate/migrate" do
+      system "dep", "ensure", "-vendor-only"
```

Comment:
Doesn't `dep` need to be installed for this command to work?

Also, any reason why `migrate` is being built from source instead of using the pre-built binaries? All modern macs should be using amd64.

Reply:
Nvm, I see `depends_on "dep" => :build` above

Reply:
> Also, any reason why migrate is being built from source instead of using the pre-built binaries?

Almost everything in Homebrew is built from source as a matter of policy. Homebrew distributes its own binaries, so the impact on users is almost nil. The upside for Homebrew of building packages from source is considerable.

Reply:
@dhui do you some idea on macos-only building and functionality testing?

Reply:
It's pretty straightforward to update the Makefile to build binaries for each platform separately.

As far as testing goes, the platform shouldn't matter but you will need docker installed.
For more info, see: https://github.com/golang-migrate/migrate/blob/master/CONTRIBUTING.md

Beware that the tests intermittently fail (sometimes the docker images don't start up in time), so I wouldn't automate anything that depends on the tests' success. Currently, I've been monitoring TravisCI builds and restarting them as necessary before each release.